src/main/java/org/opensearch/timeseries/model/Config.java

```diff
@@ -559,4 +562,14 @@ protected void checkAndThrowValidationErrors(ValidationAspect validationAspect)
             throw new TimeSeriesException(CommonMessages.FAIL_TO_VALIDATE);
         }
     }
+
+    public static Config parseConfig(Class<? extends Config> configClass, XContentParser parser) throws IOException {
+        if (configClass == AnomalyDetector.class) {
+            return AnomalyDetector.parse(parser);
+        } else if (configClass == Forecaster.class) {
+            return Forecaster.parse(parser);
+        } else {
+            throw new IllegalArgumentException("Unsupported config type");
```

Comment:
```suggestion
            throw new IllegalArgumentException("Unsupported config type. Supported config types are [AnomalyDetector, Forecaster]");
```

Reply:
Changed.

Reply:
Will you be pushing all the changes together?

Reply:
yes